src/lib/Libifl/pbsD_modify_resv.c

```diff
@@ -67,33 +67,50 @@ char *
 pbs_modify_resv(int c, char *resv_id, struct attropl *attrib, char *extend)
 {
 	struct attropl *pal = NULL;
-	int		rc = 0;
-	char		*ret = NULL;
+	int rc = 0;
+	char *ret = NULL;
+	int i;
+	svr_conn_t **svr_conns = get_conn_svr_instances(c);
+	int start = 0;
+	int ct;
+	int nsvrs = get_num_servers();
 
 	for (pal = attrib; pal; pal = pal->next)
 		pal->op = SET;
 
-	/* initialize the thread context data, if not already initialized */
-	if (pbs_client_thread_init_thread_context() != 0)
-		return NULL;
-
 	/* first verify the attributes, if verification is enabled */
-	rc = pbs_verify_attributes(c, PBS_BATCH_ModifyResv,
+	rc = pbs_verify_attributes(random_srv_conn(svr_conns), PBS_BATCH_ModifyResv,
 		MGR_OBJ_RESV, MGR_CMD_NONE, attrib);
 	if (rc)
 		return NULL;
 
-	/* lock pthread mutex here for this connection
-	 * blocking call, waits for mutex release */
-	if (pbs_client_thread_lock_connection(c) != 0)
-		return NULL;
+	if (svr_conns) {
```

Comment:
seems like there's nothing that can be done if svr_conns is NULL, so just check for if (!svr_conns) instead and return NULL

Reply:
I have added more code to this function. Please check the latest changes.